fili-core/src/main/java/com/yahoo/bard/webservice/application/AbstractBinderFactory.java

```diff
@@ -744,31 +745,76 @@ protected final void setupDruidDimensionsLoader(
      * @return The cache instance
      */
     protected DataCache<?> buildCache() {
-        if (BardFeatureFlag.DRUID_CACHE_V2.isOn()) {
-            try {
-                MemTupleDataCache<String> cache = new MemTupleDataCache<>();
-                LOG.info("MemcachedClient Version 2 started {}", cache);
-                return cache;
-            } catch (IOException e) {
-                LOG.error("MemcachedClient Version 2 failed to start {}", e);
-                throw new IllegalStateException(e);
+        if (CacheFeatureFlag.TTL.isOn() || CacheFeatureFlag.LOCAL_SIGNATURE.isOn()) { // old caching strategy
+            if (CacheFeatureFlag.LOCAL_SIGNATURE.isOn()) {
+                return buildLocalSignatureCache();
+            } else if (CacheFeatureFlag.TTL.isOn()) {
+                return buildTtlCache();
+            } else {
+                // not used, but Jersey required a binding
+                return new StubDataCache<>();
             }
-        } else if (BardFeatureFlag.DRUID_CACHE.isOn()) {
-            try {
-                DataCache<String> cache = new HashDataCache<>(new MemDataCache<HashDataCache.Pair<String, String>>());
-                LOG.info("MemcachedClient started {}", cache);
-                return cache;
-            } catch (IOException e) {
-                LOG.error("MemcachedClient failed to start {}", e);
-                throw new IllegalStateException(e);
+        } else { // new caching strategy
+            if (CacheFeatureFlag.TTL.isOn()) {
+                return buildTtlCache();
+            } else if (CacheFeatureFlag.LOCAL_SIGNATURE.isOn()) {
+                return buildLocalSignatureCache();
+            } else if (CacheFeatureFlag.ETAG.isOn()) {
+                return buildETagCahe();
+            } else {
+                return new StubDataCache<>();
             }
-        } else {
-            // not used, but Jersey required a binding
-            return new StubDataCache<>();
         }
     }
 
     /**
+     * Builds and returns an instance of local signature cache.
+     *
+     * @return the instance of local signature cache
+     */
+    private DataCache<?> buildLocalSignatureCache() {
+        LOG.warn("Cache V2 feature flag is deprecated, " +
+                "use the new configuration parameter to set desired caching strategy"
+        );
```

Comment:
TTL and localSignature could be set in new config or old config. I think there is no way to tell which config is used. In addition, LocalSignature and TTL are all marked as deprecated, which means they should always be warned, I think?